gcp/observability/config.go

```diff
@@ -72,21 +74,32 @@ func validateFilters(config *configpb.ObservabilityConfig) error {
 	return nil
 }
 
+// unmarshalConfig unmarshals a json string representing an observability config
+// into it's protobuf format.
+func unmarshalConfig(rawJSON json.RawMessage) (*configpb.ObservabilityConfig, error) {
+	var config configpb.ObservabilityConfig
+	if err := protojson.Unmarshal(rawJSON, &config); err != nil {
+		return nil, fmt.Errorf("error parsing observability config: %v", err)
+	}
+	if err := validateFilters(&config); err != nil {
+		return nil, fmt.Errorf("error parsing observability config: %v", err)
+	}
+	if config.GlobalTraceSamplingRate > 1 || config.GlobalTraceSamplingRate < 0 {
+		return nil, fmt.Errorf("error parsing observability config: invalid global trace sampling rate %v", config.GlobalTraceSamplingRate)
+	}
+	logger.Infof("Parsed ObservabilityConfig: %+v", &config)
```

Comment:
None of this is technically unmarshalling.  Move this out and make a different function? Or name this `unmarshalAndVerifyConfig`?